src/stream/src/common/table/state_table.rs

```diff
@@ -1131,6 +1215,25 @@ where
         ))
     }
 
+    pub async fn iter_with_vnode_and_output_indices(
```

Comment:
I'm worrying if this can be error-prone: one creates a state table with output indices but forget to iter with this method, or vice versa.

Is it okay to make `output_indices` an `Option`? Then we can always project with it if present, or do nothing if absent through the same entrance.